lib/ContextModule.js

```diff
@@ -1059,7 +1067,14 @@ module.exports = webpackEmptyAsyncContext;`;
 
 	getSource(sourceString) {
 		if (this.useSourceMap || this.useSimpleSourceMap) {
-			return new OriginalSource(sourceString, this.identifier());
+			return new OriginalSource(
+				sourceString,
+				makePathsRelative(
+					this.sourceOptions.context,
+					this.identifier(),
+					this.sourceOptions.cacheObj
+				)
```

Comment:
```suggestion
				`webpack://${makePathsRelative(
					this.context,
					this.identifier(),
					compilation && compilation.compiler.root
				)}`
```

Reply:
@vankop when relanding the other ContextModule PR we need to handle an unset `this.context` here, e. g. via `this.context || compilation.compiler.context`

Reply:
em...one question,why here use context module's own context to get relative path instead of using the context from options

Reply:
eg. if it will better to use compilation && compilation.option.context  ||  this.context

Reply:
module context is where nested context modules resolves. e.g.
you have files
```
/a/b/c
/a/b/b
```
then request
```js
import(`/a/b/${some}`)
```
will result in context `/a/b` (if no other resolving happen) since all submodules will be resolved related to this context => `./c`, `./b` (`/a/b/c` and `/a/b/b` respectively)

Reply:
that's right, but the case is different here, because there is module source name mapping, all module generated by webpack which name starts with `webpack://` will get its absolute path to find its module object. see,
https://github.com/webpack/webpack/blob/dfdc8b133d3d49371251e29efeefb9d31b982c9e/lib/SourceMapDevToolPlugin.js#L98

Reply:
ok, we can use only compiler context since `this.identifier()` has `context` part already..